torch/_subclasses/fake_tensor.py

```diff
@@ -2126,48 +2118,75 @@ def _check_fake_real_vals(fake: Any, real: Any) -> None:
                         f" for func: {func}"
                     ) from exc
 
-                # check fake/real tensor properies, sizes & output values
-                for i, (_real_out, _fake_out) in enumerate(
-                    zip(pytree.tree_leaves(real_out), pytree.tree_leaves(fake_out))
-                ):
-                    if isinstance(_fake_out, torch.Tensor):
+                # check fake/real tensor properies, sizes & output values,
+                # and create new fake vals if mismatched.
+                def _xref_and_maybe_replace_fake_val(path, fake, real):
+                    if isinstance(fake, torch.Tensor):
                         try:
                             _check_fake_real_tensors(
-                                _fake_out,
-                                _real_out,
+                                fake,
+                                real,
                                 context="Real tensor propagation found",
                                 sizes=False,  # manual check below
                                 strides=False,  # skip strides
                                 storage_offset=True,
                                 requires_grad=False,  # issues with FakeTensorConverter preserving requires_grad
                             )
                         except MetadataMismatchError as exc:
+                            if torch._functorch.config.generate_fake_kernels_from_real_mismatches:
+                                return _make_fake(self, func, real)
```

Comment:
Do we end up logging in the draft mode export report that there was a metadata mismatch?

Reply:
ah not yet, let me add this